tests/tests_pytorch/utilities/test_auto_restart.py

```diff
@@ -1061,429 +130,4 @@ def test_auto_restart_under_signal(on_last_batch, val_check_interval, failure_on
-            for b in batch:
-                losses.append(super().training_step(b, batch_idx)["loss"])
-            return torch.stack(losses).mean()
-
-        def validation_step(self, batch, batch_idx, dataloader_idx=0):
-            pass
-
-        validation_epoch_end = None
-
-        def _create_dataloader_kwargs(self, dataset_class, dataset_len, seed, num_workers):
-            dl_kwargs = {}
-            dl_kwargs["dataset"] = dataset_class(dataset_len, 1, seed=seed)
-            dl_kwargs["sampler"] = RandomFaultTolerantSampler(dl_kwargs["dataset"], seed=seed)
-            dl_kwargs["num_workers"] = num_workers
-            dl_kwargs["batch_size"] = 1
-            return dl_kwargs
-
-        def train_dataloader(self):
-            return [
-                DataLoader(
-                    **self._create_dataloader_kwargs(
-                        dataset_class, 10, seed, seed + 1 if val_check_interval == 1.0 else 0
-                    )
-                )
-                for seed, dataset_class in enumerate(train_dataset_cls)
-            ]
-
-        def val_dataloader(self):
-            return [
-                DataLoader(**self._create_dataloader_kwargs(dataset_class, 1, seed, 0))
-                for seed, dataset_class in enumerate(val_dataset_cls)
-            ]
-
-        def configure_optimizers(self):
-            optimizer = torch.optim.SGD(self.layer.parameters(), lr=0.001)
-            lr_scheduler = torch.optim.lr_scheduler.StepLR(optimizer, step_size=1)
-            return [optimizer], [lr_scheduler]
-
-    seed_everything(42)
-    model = TestModel()
-    trainer = Trainer(default_root_dir=tmpdir, max_epochs=1, val_check_interval=val_check_interval)
-    trainer.fit(model)
-    total_batches = model.batches
-    total_weight = deepcopy(model.layer.weight)
-    trainer.train_dataloader = None
-
-    seed_everything(42)
-    model = TestModel(should_fail=True)
-    trainer = Trainer(
-        default_root_dir=tmpdir,
-        max_epochs=1,
-        val_check_interval=val_check_interval,
-        callbacks=OnExceptionCheckpoint(tmpdir),
-    )
-    with pytest.raises(CustomException):
-        trainer.fit(model)
-    trainer.train_dataloader = None
-    failed_batches = model.batches
-    failed_weight = deepcopy(model.layer.weight)
-
-    checkpoint_path = str(tmpdir / "on_exception.ckpt")
-    assert os.path.exists(checkpoint_path)
-
-    seed_everything(42)
-    model = TestModel()
-    trainer = Trainer(default_root_dir=tmpdir, max_epochs=1, val_check_interval=val_check_interval)
-    trainer.fit(model, ckpt_path=checkpoint_path)
-    trainer.train_dataloader = None
-    restart_batches = model.batches
-
-    torch.testing.assert_close(total_batches, failed_batches + restart_batches)
-    assert not torch.equal(total_weight, failed_weight)
-    assert torch.equal(total_weight, model.layer.weight)
+    _fit_model(tmpdir, True, val_check_interval, failure_on_step, failure_on_training, on_last_batch, status=status)
```

Comment:
This tests where in the loops we check if the SIGTERM signal was received to raise an exception. It could be moved to signal connector or a loops test file.